pkg/asset/machines/azure/azuremachines.go

```diff
@@ -0,0 +1,251 @@
+		id := clusterID
+		if hyperVGen == "V2" {
+			id += "-gen2"
+		}
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/latest", resourceGroup, galleryName, id)
+		image.ID = &imageID
+	} else {
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/images/%s", resourceGroup, clusterID)
+		if hyperVGen == "V2" && platform.CloudName != azure.StackCloud {
+			imageID += "-gen2"
+		}
+		image.ID = &imageID
+	}
+
+	osDisk := capz.OSDisk{
+		OSType:     "Linux",
+		DiskSizeGB: &mpool.DiskSizeGB,
+		ManagedDisk: &capz.ManagedDiskParameters{
+			StorageAccountType: mpool.DiskType,
+		},
+	}
+	ultrassd := mpool.UltraSSDCapability == "Enabled"
+	additionalCapabilities := &capz.AdditionalCapabilities{
+		UltraSSDEnabled: &ultrassd,
+	}
+
+	machineProfile := generateSecurityProfile(mpool)
+	securityProfile := &capz.SecurityProfile{
+		EncryptionAtHost: machineProfile.EncryptionAtHost,
+		SecurityType:     capz.SecurityTypes(machineProfile.Settings.SecurityType),
+	}
+	if machineProfile.Settings.ConfidentialVM != nil {
+		securityProfile.UefiSettings = &capz.UefiSettings{
+			VTpmEnabled:       ptr.To(machineProfile.Settings.ConfidentialVM.UEFISettings.VirtualizedTrustedPlatformModule == "Enabled"),
+			SecureBootEnabled: ptr.To(machineProfile.Settings.ConfidentialVM.UEFISettings.SecureBoot == "Enabled"),
+		}
+	} else if machineProfile.Settings.TrustedLaunch != nil {
+		securityProfile.UefiSettings = &capz.UefiSettings{
+			VTpmEnabled:       ptr.To(machineProfile.Settings.TrustedLaunch.UEFISettings.VirtualizedTrustedPlatformModule == "Enabled"),
+			SecureBootEnabled: ptr.To(machineProfile.Settings.TrustedLaunch.UEFISettings.SecureBoot == "Enabled"),
+		}
+	}
+
+	providerID := fmt.Sprintf("%s-identity", clusterID)
+	var result []*asset.RuntimeFile
+	for idx := int64(0); idx < total; idx++ {
+		zone := mpool.Zones[int(idx)%len(mpool.Zones)]
+		azureMachine := &capz.AzureMachine{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+				Kind:       "AzureMachine",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-%s-%d", clusterID, pool.Name, idx),
+				Labels: map[string]string{
+					"cluster.x-k8s.io/control-plane": "",
+					"cluster.x-k8s.io/cluster-name":  clusterID,
+				},
+			},
+			Spec: capz.AzureMachineSpec{
+				ProviderID:             ptr.To(providerID),
+				VMSize:                 mpool.InstanceType, // required
+				FailureDomain:          ptr.To(zone),
+				OSDisk:                 osDisk, // required
+				SSHPublicKey:           "",
+				AdditionalTags:         tags,
+				AdditionalCapabilities: additionalCapabilities,
+				AllocatePublicIP:       false,
+				EnableIPForwarding:     false,
+			},
+		}
+		*azureMachine.Spec.Image = *image
+		*azureMachine.Spec.AdditionalCapabilities = *additionalCapabilities
+		*azureMachine.Spec.SecurityProfile = *securityProfile
```

Comment:
Let's move these into the literal, so they are set like the other fields immediately above this

Reply:
Wouldn't the mutable nature cause problems? I had some problems back in the CPMS manifests creation, hence the DeepCopy before although I don't see them changing. Or are you suggesting something else?

Reply:
Shouldn't be a problem. My suggestion should be the equivalent of what you're doing now, except instead of setting the values in two different ways, all of the values are being set in the [struct literal](https://go.dev/tour/moretypes/5). I tried to illustrate the idea here: https://github.com/openshift/installer/pull/7969/files#r1488727469

Reply:
Ok it's the same thing. We had a problem in CPMS where since all these variables have the same address, change in one of these resulted in all of them changing so we had to use DeepCopy/*var=*var. The change in value happened in the CPMS operator and if the CAPI operator changed this, it would be problematic.